contracts/stellar-interchain-token-service/src/migrate.rs

```diff
@@ -68,6 +71,32 @@ pub fn migrate_token(
         token_manager_type,
     } = storage::try_token_id_config(env, token_id.clone()).ok_or(ContractError::InvalidTokenId)?;
 
+    env.authorize_as_current_contract(vec![
+        env,
+        InvokerContractAuthEntry::Contract(SubContractInvocation {
+            context: ContractContext {
+                contract: token_manager.clone(),
```

Comment:
the first call is to the upgrader's `upgrade` function so should this be at a level lower? As long as it matches `env.auths()` it should be fine